### PR TITLE
Feature/refunds promo acct level

### PR DESCRIPTION
**Pull request by ps2-controller**
This PR does two things
- makes refunds endpoint
- makes promo_len, promo_purchase_window_len, and promo_min_pay_percent available at the account level.

### CODE REVIEW

**Comment by dangdennis** [comment]
@ps2-controller can you link the linear ticket in the title?

**Comment by ps2-controller** [comment]
Linear, from BHG board:
Refunds:
https://linear.app/canopy-bhg/issue/BHG-18/refunds-api-endpoint

acct level params:
https://linear.app/canopy-bhg/issue/BHG-22/make-promo-min-pay-percent-overwritable-at-account-level

File: openapi.json [hunk 1]
```diff
@@ -1229,6 +1229,51 @@
...
+          {
+            "name": "account_id",
+            "in": "path",
+            "description": "Account ID",
+            "required": true,
+            "style": "simple",
+            "explode": false,
+            "schema": {
+              "type": "string",
+              "example": "31mNprzLd2bKl6koVna68ARM"
```
Comment (dangdennis) [inline openapi.json:1248]: we should use a UUID as the example now 
```suggestion
              "example": "f3991696-ad13-41bf-864d-178a8079409b"
```

File: schemas/CreateAccountInput.json [hunk 2]
```diff
@@ -119,6 +119,22 @@
         "promo_overview": {
           "type": "object",
           "properties": {
+            "promo_len": {
+              "type": "integer",
+              "description": "If applicable, promo length in cycles. If not provided, the product promo_len will be used.",
```
Comment (dangdennis) [inline schemas/CreateAccountInput.json:124]:
```suggestion
              "description": "If applicable, promo length in cycles. If not provided, defaults to the product promo_len.",
```

File: schemas/CreateAccountInput.json [hunk 3]
```diff
@@ -119,6 +119,22 @@
...
+              "example": 500
+            },
+            "promo_purchase_window_len": {
+              "type": "integer",
+              "description": "Only for multi-advance installment loans with a purchase window. If applicable, promo length in cycles. If not provided, the product promo_len will be used.",
```
Comment (dangdennis) [inline schemas/CreateAccountInput.json:129]:
```suggestion
              "description": "Only for multi-advance installment loans with a purchase window. If applicable, promo length in cycles. If not provided, defaults to the product promo_len.",
```

File: schemas/CreateAccountInput.json [hunk 4]
```diff
@@ -119,6 +119,22 @@
...
+              "example": 500
+            },
+            "promo_min_pay_percent": {
+              "type": "number",
+              "description": "Defaults to product `promo_min_pay_percent`. The percent of the interest calculation type used. For example, if `PERCENT_INTEREST` is the interest calculation type, and this value is 50, 50% of interest accrued during the cycle will be reflected as the required minimum payment.",
```
Comment (dangdennis) [inline schemas/CreateAccountInput.json:134]: here you use backticks to highlight promo_min_pay_percent. but you didn't do it for promo_len. can you pick one?

Comment (dangdennis) [reply schemas/CreateAccountInput.json:134]: cool feature

File: schemas/CreateRefundInput.json [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+          "value": {
+            "type": "string",
+            "description": "Value - i.e. External Account ID",
+            "example": "22445702-a389-431f-927d-07b8d0750787"
+          }
+        }
+      }
+    }
+  }
+}
```
Comment (dangdennis) [inline schemas/CreateRefundInput.json:71]: isn't there a way to reuse external_fields and merchant_data schema? as well as line item status?

Comment (ps2-controller) [reply schemas/CreateRefundInput.json:71]: Yeah we can probably separate "external_fields" out into its own schema and reference it where used. Would be a good project for Alex, I'll pair with him on it separately

Comment (ps2-controller) [reply schemas/CreateRefundInput.json:71]: same w/ merchant_data

File: schemas/CreateAccountInput.json (hunk 4 above)
Comment (ps2-controller) [reply schemas/CreateAccountInput.json:134]: yep, added backticks to `promo_len`

**Review by dangdennis — APPROVED** [review]

**Pull request merged**